### PR TITLE
Add module for NASHA management

### PR DESCRIPTION
**Pull request by eliesmb**
Add an Ansible module to manage OVH NASHA creation, its ACL and Sapshots management too.

### CODE REVIEW

File: plugins/modules/dedicated_nasha_manage_partition.py [hunk 1]
```diff
@@ -0,0 +1,534 @@
...
+    module.exit_json(
+        msg=final_message + "{0}\n".format(
+            DRY_RUN_MSG
+        ),
+        changed=all_changed,
+    )
+
+
+def main():
+    if not HAS_OVH:
```
Comment (Skunnyk) [inline plugins/modules/dedicated_nasha_manage_partition.py:527]: This is handled by `module_utils/ovh.py`

**Comment by Skunnyk** [comment]
Hi, we don't use NASHA so I can't test your work. I suppose you tested it and the behavior is correct ? ;)

File: plugins/modules/dedicated_nasha_manage_partition.py (hunk 1 above)
Comment (eliesmb) [reply plugins/modules/dedicated_nasha_manage_partition.py]: Hello, I removed it from the module, thanks

**Comment by eliesmb** [comment]
> Hi, we don't use NASHA so I can't test your work. I suppose you tested it and the behavior is correct ? ;)

Hi, I just added a fix about ACL management. Now, everything work correctly.

**Comment by guillaume-gambs** [comment]
@Skunnyk we've been using it in production for a few months now.


Can you validate the PR so that we can go back to your repository?

**Review by carrouet — APPROVED** [review]

**Review by tortuegenialez — APPROVED** [review]

**Comment by guillaume-gambs** [comment]
@Skunnyk @tortuegenialez @carrouet 

Hello Guys, is there anything that can help you to merge the pull request ?

**Comment by Skunnyk** [comment]
Hi!

Can you rework the module based on https://github.com/synthesio/infra-ovh-ansible-module/commit/2b6b879d56b247673fdb6cb9b46502b72467246e ? (no need to handle exceptions in all modules, this is not factorized).
About ansible-galaxy, I need to take a look :)

File: .flake8 [hunk 2]
```diff
@@ -1,5 +1,6 @@
 [flake8]
 # https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html
 # Ignore "Module level import not at top of file (E402)"
-ignore=E402
+# Use `docker run -ti --rm -v $(pwd):/apps alpine/flake8 -v plugins/modules/*` to validate
+ignore=E402,W504,W503
```
Comment (Skunnyk) [inline .flake8:5]: Why is this needed ?

Comment (guillaume-gambs) [reply .flake8:5]: I don't like warning, and this warning is in conflict.
https://www.flake8rules.com/rules/W503.html
https://www.flake8rules.com/rules/W504.html
maybe we can keep check on w504 and ignore W503

Comment (Skunnyk) [reply .flake8:5]: I don't like warnings neither, but… I don't see any warnings :D

Comment (guillaume-gambs) [reply .flake8:5]: I have this warning : 

```
plugins/modules/dedicated_nasha_manage_partition.py:288:17: W503 line break before binary operator
plugins/modules/dedicated_nasha_manage_partition.py:334:30: W503 line break before binary operator
plugins/modules/dedicated_nasha_manage_partition.py:339:17: W503 line break before binary operator
plugins/modules/dedicated_nasha_manage_partition.py:347:13: W503 line break before binary operator
plugins/modules/dedicated_nasha_manage_partition.py:386:25: W503 line break before binary operator
plugins/modules/dedicated_nasha_manage_partition.py:462:13: W503 line break before binary operator
plugins/modules/dedicated_nasha_manage_partition.py:467:9: W503 line break before binary operator
plugins/modules/dedicated_nasha_manage_partition.py:468:9: W503 line break before binary operator
```

**Comment by guillaume-gambs** [comment]
@Skunnyk all is good for you ?

File: .flake8 (hunk 2 above)
Comment (Skunnyk) [reply .flake8]: Meh, I don't see anything :thinking: 
```bash
$ docker run -ti --rm -v $(pwd):/apps alpine/flake8 --version
7.0.0 (flake8-sarif: 1.0.15, mccabe: 0.7.0, pycodestyle: 2.11.1, pyflakes: 3.2.0) CPython 3.12.3 on Linux
$ cat .flake8 
[flake8]
# https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html
# Ignore "Module level import not at top of file (E402)"
ignore=E402
max-line-length = 150
 $ docker run -ti --rm -v $(pwd):/apps alpine/flake8 -v plugins/modules/*
flake8.checker            MainProcess    169 INFO     Making checkers
flake8.main.application   MainProcess    238 INFO     Finished running
flake8.main.application   MainProcess    238 INFO     Reporting errors
flake8.main.application   MainProcess    238 INFO     Found a total of 33 violations and reported 0
```

The 33 violations are from E402 (ignored)

Comment (guillaume-gambs) [reply .flake8]: on which branch ?

**Review by Skunnyk — APPROVED** [review]

**Comment by Skunnyk** [comment]
Aaand thank you :)
